### PR TITLE
feat: added navbar component

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
feat: added navbar component

### CODE REVIEW

File: src/components/ConfigProvider/Theming/styleGenerator.ts [hunk 1]
```diff
@@ -126,6 +126,14 @@ export function getStyle(themeOptions: ThemeOptions): IGetStyle {
         };
     }
 
+    // ================= Navbar theme ==================
+    if (theme.navbarTheme) {
+        variables = {
+            ...variables,
+            ...themeGenerator(theme.tabsTheme, 'navbar'),
```
Comment (ayadav-eightfold) [inline src/components/ConfigProvider/Theming/styleGenerator.ts:133]: change

Comment (ayadav-eightfold) [reply src/components/ConfigProvider/Theming/styleGenerator.ts:133]: add story in configProvider

File: src/components/Navbar/navbar.module.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+.navbar-container {
+    --navbar-background: var(--primary-color-80);
```
Comment (ayadav-eightfold) [inline src/components/Navbar/navbar.module.scss:2]: add to default theme

File: src/components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { Ref, FC } from 'react';
+
+// Styles:
+import styles from './navbar.module.scss';
+import { NavbarProps } from './';
+import { mergeClasses } from '../../shared/utilities';
+
+export const NavbarContent: FC<NavbarProps> = React.forwardRef(
```
Comment (ayadav-eightfold) [inline src/components/Navbar/Navbar.tsx:8]: New file

File: src/styles/themes/_default-theme.scss [hunk 4]
```diff
@@ -137,4 +137,31 @@
...
+    --navbar-text: var(--text-inverse-color);
+    --navbar-icon: var(--navbar-text);
+    --navbar-active-background: var(--navbar-background);
+    --navbar-active-text: var(--navbar-text);
+    --navbar-active-icon: var(--navbar-text);
+    --navbar-search-background: var(--navbar-background);
+    --navbar-search-text: var(--grey-color-40);
+    --navbar-search-border: var(--border-color);
+    --navbar-search-icon: var(--grey-color-40);
+    --navbar-search-placeholder: var(--grey-color-60);
```
Comment (ayadav-eightfold) [inline src/styles/themes/_default-theme.scss:152]: rename and move to searchbar

File: src/styles/themes/_default-theme.scss [hunk 5]
```diff
@@ -137,4 +137,31 @@
...
+    --navbar-search-active-background: var(--background-color);
+    --navbar-search-active-text: var(--text-secondary-color);
+    --navbar-search-category-background: var(--grey-color-10);
+    --navbar-search-category-text: var(--grey-color-80);
+    --navbar-search-hover-background: var(--background-color);
+    --navbar-search-hover-text: var(--navbar-background);
+    --navbar-search-hover-icon: var(--navbar-background);
+    --navbar-dropdown-background: var(--background-color);
+    --navbar-dropdown-text: var(--text-secondary-color);
```
Comment (ayadav-eightfold) [inline src/styles/themes/_default-theme.scss:161]: move to dropdown, panel, modal

File: src/components/Navbar/Navbar.types.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { OcBaseProps } from '../OcBase';
+import { Value } from '../ConfigProvider';
+
+export interface NavbarProps extends OcBaseProps<HTMLDivElement> {}
+
+export interface NavbarTheme {
+    background?: Value;
+    textColor?: Value;
+    hoverBackground?: Value;
+    hoverTextColor?: Value;
```
Comment (ychhabra-eightfold) [inline src/components/Navbar/Navbar.types.ts:10]:
```suggestion
    textHoverBackground?: Value;
    textGoverColor?: Value;
```

File: src/styles/themes/_default-theme.scss [hunk 7]
```diff
@@ -137,4 +137,24 @@
...
+    --searchbar-color: var(--grey-color-40);
+    --searchbar-border: var(--border-color);
+    --searchbar-icon-color: var(--grey-color-40);
+    --searchbar-placeholder: var(--grey-color-60);
+    --searchbar-active-background: var(--background-color);
+    --searchbar-active-color: var(--text-secondary-color);
+    --searchbar-item-hover-background: var(--background-color);
+    --searchbar-item-hover-color: var(--navbar-background);
+    --searchbar-item-hover-icon-color: var(--navbar-background);
+    /* ------ Searchbar theme ------ */
```
Comment (ychhabra-eightfold) [inline src/styles/themes/_default-theme.scss:159]: remove these for now

File: src/components/ConfigProvider/ConfigProvider.stories.tsx [hunk 8]
```diff
@@ -276,6 +316,10 @@ Theming.args = {
                 pillHoverLabel: '--primary-color',
                 pillBackground: '--grey-color-10',
             },
+            navbarTheme: {
+                background: '--primary-color-80',
+                textColor: '--primary-color-10',
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.stories.tsx:321]: add other props as well so its easier to see in sotrybook

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

**Pull request merged**